src/typesystem.cpp

```diff
@@ -246,3 +246,31 @@ MonoId TypeSystemCore::inst_fresh(PolyId poly) {
 		vals.push_back(new_var());
 	return inst_with(poly, vals);
 }
+
+TypeFunctionId TypeSystemCore::func_find(TypeFunctionId func) {
+	TypeFunctionData& func_data = type_function_data[func];
+
+	if (func_data.equals == func or
+	    func_data.type == type_function_type::Known)
```

Comment:
NIT: this check should be backwards.

I know it is equivalent, but checking `.equals` when the func is known makes no sense.